app/src/test/java/org/dhis2/usescases/enrollment/EnrollmentPresenterImplTest.kt

```diff
@@ -246,4 +247,86 @@ class EnrollmentPresenterImplTest {
 
         verify(enrollmentView).setResultAndFinish()
     }
+
```

Comment:
In the test we like to user three blocks (preparation, execution and verification).

Please Try reduce the code inside of every test breaking down in functions.
For example:
givenATei (status)
givenAprogram()

In these functions you create TEI, program and stub functions in repositories returning the TEI or program.

Using this functions you can avoid the duplicate code that exist currently in these tests.